flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/SearchOperatorGen.scala

```diff
@@ -82,14 +82,14 @@ object SearchOperatorGen {
         .map(CastRuleProvider.cast(toCastContext(ctx), sargType, commonType, _))
         .map(generateLiteral(ctx, _, commonType))
       if (sarg.containsNull) {
-        haystack += generateNullLiteral(commonType, ctx.nullCheck)
+        haystack += generateNullLiteral(commonType)
       }
       val setTerm = ctx.addReusableHashSet(haystack.toSeq, commonType)
       val negation = if (sarg.isComplementedPoints) "!" else ""
 
       val Seq(resultTerm, nullTerm) = newNames("result", "isNull")
 
-      val operatorCode = if (ctx.nullCheck) {
+      val operatorCode = if (commonType.isNullable) {
```

Comment:
This is not well tested. `nullTerm` would not be declared and fail the compilation.

Reply:
thank you, removed!